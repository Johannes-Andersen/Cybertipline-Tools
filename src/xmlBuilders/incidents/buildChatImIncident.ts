import { XMLBuilder } from 'fast-xml-parser';
import type { ChatImIncident } from '../../types';

const builder = new XMLBuilder({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  suppressBooleanAttributes: false,
});

export const buildChatImIncident = (
  chatImIncident: ChatImIncident,
  keyName = 'chatImIncident',
): string =>
  builder.build({
    [keyName]: {
      chatClient: chatImIncident.chatClient,
      chatRoomName: chatImIncident.chatRoomName,
      content: chatImIncident.content,
      additionalInfo: chatImIncident.additionalInfo,
    },
  });
