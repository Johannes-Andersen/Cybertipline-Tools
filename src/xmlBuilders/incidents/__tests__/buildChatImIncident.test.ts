import { describe, expect, it } from 'vitest';
import type { ChatImIncident } from '../../../types';
import { buildChatImIncident } from '../buildChatImIncident';

describe('buildChatImIncident', () => {
  it('should build a chat im incident', () => {
    const chatImIncident: ChatImIncident = {
      additionalInfo: 'additionalInfo',
      chatRoomName: 'chatRoomName',
      chatClient: 'chatClient',
      content: 'content',
    };

    const result = buildChatImIncident(chatImIncident);

    expect(result).toMatchSnapshot();
  });
});
