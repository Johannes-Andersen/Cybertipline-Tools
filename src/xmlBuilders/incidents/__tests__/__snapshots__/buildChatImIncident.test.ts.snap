// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildChatImIncident > should build a chat im incident 1`] = `"<chatImIncident><chatClient>chatClient</chatClient><chatRoomName>chatRoomName</chatRoomName><content>content</content><additionalInfo>additionalInfo</additionalInfo></chatImIncident>"`;
